deployment/cdk/opensearch-service-migration/lib/stack-composer.ts

```diff
@@ -202,17 +242,45 @@ export class StackComposer {
         const captureProxyExtraArgs = this.getContextForType('captureProxyExtraArgs', 'string', defaultValues, contextJSON)
         const elasticsearchServiceEnabled = this.getContextForType('elasticsearchServiceEnabled', 'boolean', defaultValues, contextJSON)
         const kafkaBrokerServiceEnabled = this.getContextForType('kafkaBrokerServiceEnabled', 'boolean', defaultValues, contextJSON)
-        const targetClusterEndpoint = this.getContextForType('targetClusterEndpoint', 'string', defaultValues, contextJSON)
         const fetchMigrationEnabled = this.getContextForType('fetchMigrationEnabled', 'boolean', defaultValues, contextJSON)
         const dpPipelineTemplatePath = this.getContextForType('dpPipelineTemplatePath', 'string', defaultValues, contextJSON)
-        const sourceClusterDisabled = this.getContextForType('sourceClusterDisabled', 'boolean', defaultValues, contextJSON)
-        const sourceClusterEndpoint = this.getContextForType('sourceClusterEndpoint', 'string', defaultValues, contextJSON)
         const osContainerServiceEnabled = this.getContextForType('osContainerServiceEnabled', 'boolean', defaultValues, contextJSON)
         const otelCollectorEnabled = this.getContextForType('otelCollectorEnabled', 'boolean', defaultValues, contextJSON)
         const reindexFromSnapshotServiceEnabled = this.getContextForType('reindexFromSnapshotServiceEnabled', 'boolean', defaultValues, contextJSON)
         const reindexFromSnapshotExtraArgs = this.getContextForType('reindexFromSnapshotExtraArgs', 'string', defaultValues, contextJSON)
         const albAcmCertArn = this.getContextForType('albAcmCertArn', 'string', defaultValues, contextJSON);
 
+        const sourceClusterDisabled = this.getContextForType('sourceClusterDisabled', 'boolean', defaultValues, contextJSON)
```

Comment:
These are all top level fields, what would be the level of effort to consolidate it into an object/nested object like we have in sourceClusterAuth.

e.g.
```ts
{
   "sourceCluster": {
      "auth": {
         "type": "sigv4"
         "region": "us-west-2"
         "serviceSigningName": "es"
      },
      "endpoint": "https://source-cluster:9200",
      "version" : "ES_7.10"
      "disabled": false // optional/inferred
}
```

Reply:
Maybe something like this:

```ts
interface BasicAuth {
  type: 'basic';
  username: string;
  password?: string;
  passwordFromArn?: string;
}

interface SigV4Auth {
  type: 'sigv4';
  region: string;
  serviceSigningName: string;
}

interface NoAuth {
  type: 'none';
}

// Union type for all auth types
type Auth = BasicAuth | SigV4Auth | NoAuth;

// Function to parse and validate auth object
function parseAuth(json: any): Auth | null {
  if (json.type === 'basic' && typeof json.username === 'string' && (typeof json.password === 'string' || typeof json.passwordFromArn === 'string') && !(typeof json.password === 'string' && typeof json.passwordFromArn === 'string')) {
    return json as BasicAuth;
  } else if (json.type === 'sigv4' && typeof json.region === 'string' && typeof json.serviceSigningName === 'string') {
    return json as SigV4Auth;
  } else if (json.type === 'none') {
    return json as NoAuth;
  } else {
    return null; // Invalid auth type
  }
}
```

Reply:
**Offline Discussion**

Discussed how we can address both concerns with keeping this as a single object while not immediately removing the flattened fields from the cdk.context.json to retain some compatibility with both ways of specifying for a short period.